cmd/kubeadm/app/util/endpoint.go

```diff
@@ -32,17 +32,17 @@ import (
 // - If the ControlPlaneEndpoint is defined, use it.
 // - if the ControlPlaneEndpoint is defined but without a port number, use the ControlPlaneEndpoint + api.BindPort is used.
 // - Otherwise, in case the ControlPlaneEndpoint is not defined, use the api.AdvertiseAddress + the api.BindPort.
-func GetMasterEndpoint(cfg *kubeadmapi.InitConfiguration) (string, error) {
+func GetMasterEndpoint(cfg *kubeadmapi.ClusterConfiguration, localEndpoint *kubeadmapi.APIEndpoint) (string, error) {
```

Comment:
What do you think about replacing `cfg ClusterConfiguration` with `ControlPlaneEndpoint  string`?`